src/main/kotlin/uk/gov/gdx/datashare/services/ScheduledConsumedEventCheckerService.kt

```diff
@@ -0,0 +1,20 @@
+package uk.gov.gdx.datashare.services
+
+import net.javacrumbs.shedlock.core.LockAssert
+import net.javacrumbs.shedlock.spring.annotation.SchedulerLock
+import org.springframework.scheduling.annotation.Scheduled
+import org.springframework.stereotype.Service
+import java.util.concurrent.TimeUnit
+
+@Service
+class ScheduledConsumedEventCheckerService(
+  private val eventConsumedCheckingService: EventConsumedCheckingService,
+) {
+
+  @Scheduled(fixedRate = 1, timeUnit = TimeUnit.HOURS)
+  @SchedulerLock(name = "checkConsumedEvents", lockAtMostFor = "3m", lockAtLeastFor = "3m")
```

Comment:
Are these the lock time periods we want? I'm unsure of how to calculate them

Reply:
Not sure what we want yet, probs need to think about how long to keep lock open for and how often to run